pandas/tests/test_resample.py

```diff
@@ -783,15 +783,21 @@ def test_resample_empty_dataframe(self):
 
         for freq in ['M', 'D', 'H']:
             # count retains dimensions too
-            methods = downsample_methods + ['count']
+            methods = downsample_methods + upsample_methods
             for method in methods:
                 result = getattr(f.resample(freq), method)()
+                if method != 'size':
+                    expected = f.copy()
+                    assert_type_equal = assert_frame_equal
+                else:
+                    # GH14962
+                    expected = Series([])
+                    assert_type_equal = assert_series_equal
 
-                expected = f.copy()
                 expected.index = f.index._shallow_copy(freq=freq)
                 assert_index_equal(result.index, expected.index)
                 assert result.index.freq == expected.index.freq
-                assert_frame_equal(result, expected, check_dtype=False)
+                assert_type_equal(result, expected, check_dtype=False)
```

Comment:
you can use ``assert_almost_equal`` here (it dispatches) rather than rolling your own